lib/mongoidable/current_ability.rb

```diff
@@ -8,18 +8,71 @@ module Mongoidable
   #   own static class abilities (including base class abilities)
   #   own instance abilities
   module CurrentAbility
+    extend Memoist
     attr_accessor :parent_model
 
-    def current_ability(parent = nil)
-      @abilities ||= Mongoidable::Abilities.new(mongoidable_identity, parent || self)
-      @abilities.reset
-      add_inherited_abilities
-      add_ancestral_abilities(parent)
-      @abilities.merge(own_abilities)
+    def current_ability(parent = @parent_model || nil)
+      @parent_model ||= parent
+      if @abilities.blank? || @renew
+        @abilities = @abilities&.empty_clone || Mongoidable::Abilities.new(mongoidable_identity, @parent_model || self)
+        run_before_ability_callbacks(:inherited)
+        @abilities.merge(inherited_abilities(@renew_inherited))
+        run_after_ability_callbacks(:inherited)
+
+        run_before_ability_callbacks(:ancestral)
+        @abilities.merge(ancestral_abilities(self.changed? || @renew_ancestral))
```

Comment:
I'm not seeing where `ancestral_abilities` is defined to take a parameter.  I see definitions without a parameter, but none with one.

I could be missing it, but if so, I could use some help finding it.

Reply:
```
Every memoized function (which initially was not accepting any arguments) has a (reload) argument you can pass in to bypass and reset the memoization:

def some_method
  Time.now
end
memoize :some_method
```